libkbfs/bare_root_metadata_test_util.go

```diff
@@ -15,33 +17,42 @@ import (
 // BareRootMetadata objects don't have enough data to build a
 // TlfHandle from until the first rekey. pubKey is non-empty only for
 // server-side tests.
-func FakeInitialRekey(md MutableBareRootMetadata, codec kbfscodec.Codec,
-	crypto cryptoPure, h tlf.Handle,
-	pubKey kbfscrypto.TLFPublicKey) ExtraMetadata {
-	var readerEPubKeyIndex int
-	// Apply the "negative hack" for V2 and earlier.
-	if md.Version() <= InitialExtraMetadataVer {
-		readerEPubKeyIndex = -1
+func FakeInitialRekey(md MutableBareRootMetadata,
+	h tlf.Handle, pubKey kbfscrypto.TLFPublicKey) ExtraMetadata {
+	if md.LatestKeyGeneration() >= FirstValidKeyGen {
+		panic(fmt.Errorf("FakeInitialRekey called on MD with existing key generations"))
 	}
-	wDkim := make(UserDeviceKeyInfoMap)
+
+	wKeys := make(UserDevicePublicKeys)
 	for _, w := range h.Writers {
 		k := kbfscrypto.MakeFakeCryptPublicKeyOrBust(string(w))
-		wDkim[w] = DeviceKeyInfoMap{
-			k: TLFCryptKeyInfo{},
+		wKeys[w] = DevicePublicKeys{
+			k: true,
 		}
 	}
 
-	rDkim := make(UserDeviceKeyInfoMap)
+	rKeys := make(UserDevicePublicKeys)
 	for _, r := range h.Readers {
 		k := kbfscrypto.MakeFakeCryptPublicKeyOrBust(string(r))
-		rDkim[r] = DeviceKeyInfoMap{
-			k: TLFCryptKeyInfo{
-				EPubKeyIndex: readerEPubKeyIndex,
-			},
+		rKeys[r] = DevicePublicKeys{
+			k: true,
 		}
 	}
 
+	codec := kbfscodec.NewMsgpack()
+	crypto := MakeCryptoCommon(codec)
```

Comment:
Done, although I don't know if we'll ever actually need to pass something in other than the usual CryptoCommon.